src/main/java/org/kohsuke/github/GHWorkflowRunJob.java

```diff
@@ -0,0 +1,256 @@
+     *
+     * @return completion date
+     */
+    public Date getCompletedAt() {
+        return GitHubClient.parseDate(completedAt);
+    }
+
+    /**
+     * Gets status of the job.
+     * <p>
+     * Can be {@code UNKNOWN} if the value returned by GitHub is unknown from the API.
+     *
+     * @return status of the job
+     */
+    public Status getStatus() {
+        return Status.from(status);
+    }
+
+    /**
+     * Gets the conclusion of the job.
+     * <p>
+     * Can be {@code UNKNOWN} if the value returned by GitHub is unknown from the API.
+     *
+     * @return conclusion of the job
+     */
+    public Conclusion getConclusion() {
+        return Conclusion.from(conclusion);
+    }
+
+    /**
+     * The run id.
+     *
+     * @return the run id
+     */
+    public long getRunId() {
+        return runId;
+    }
+
+    @Override
+    public URL getHtmlUrl() {
+        return GitHubClient.parseURL(htmlUrl);
+    }
+
+    /**
+     * The check run URL.
+     *
+     * @return the check run url
+     */
+    public URL getCheckRunUrl() {
+        return GitHubClient.parseURL(checkRunUrl);
+    }
+
+    /**
+     * Gets the execution steps of this job.
+     *
+     * @return the execution steps
+     */
+    public List<Step> getSteps() {
+        return steps;
+    }
+
+    /**
+     * Repository to which the job belongs.
+     *
+     * @return the repository
+     */
+    public GHRepository getRepository() {
+        return owner;
+    }
+
+    /**
+     * Downloads the logs.
+     * <p>
+     * The logs are returned as a text file.
+     *
+     * @param <T>
+     *            the type of result
+     * @param streamFunction
+     *            The {@link InputStreamFunction} that will process the stream
+     * @throws IOException
+     *             The IO exception.
+     * @return the result of reading the stream.
+     */
+    public <T> T downloadLogs(InputStreamFunction<T> streamFunction) throws IOException {
+        requireNonNull(streamFunction, "Stream function must not be null");
+
+        return root.createRequest().method("GET").withUrlPath(getApiRoute(), "logs").fetchStream(streamFunction);
+    }
+
+    private String getApiRoute() {
+        if (owner == null) {
+            // Workflow runs returned from search to do not have an owner. Attempt to use url.
+            final URL url = Objects.requireNonNull(getUrl(), "Missing instance URL!");
+            return StringUtils.prependIfMissing(url.toString().replace(root.getApiUrl(), ""), "/");
+
+        }
+        return "/repos/" + owner.getOwnerName() + "/" + owner.getName() + "/actions/jobs/" + getId();
+    }
+
+    GHWorkflowRunJob wrapUp(GHRepository owner) {
+        this.owner = owner;
+        return wrapUp(owner.root);
+    }
+
+    GHWorkflowRunJob wrapUp(GitHub root) {
+        this.root = root;
+        if (owner != null) {
+            owner.wrap(root);
+        }
+        return this;
+    }
+
+    public static class Step {
```

Comment:
I decided to go for a simple inline class here. Can change it if you prefer something else.

Reply:
I have no particular objection to inner classes when they make sense.  This seems reasonable.  What was your concern here?

Reply:
Well, it's always hard to predict if at some point they will add more stuff into this and it would have made sense to make it a root level object. But I think we are relatively safe here.